learning/tour-of-beam/learning-content/java/schema-based-transforms/coder/description.md

```diff
@@ -0,0 +1,126 @@
+
+When you create custom objects and schemas, you need to create a subclass of Coder for your object and implement the following methods:
+* `encode` - converting objects to bytes
+* `decode` - converting bytes to objects
+* `getCoderArguments` - If it is a `Coder` for a parameterized type, returns a list of `Coders` used for each of the parameters, in the same order in which they appear in the type signature of the parameterized type.
+* `verifyDeterministic` - throw the `Coder.NonDeterministicException`, if the encoding is not deterministic.
+
+When you get the data and when you paint it as a structure, you will need a `dto` class. In this case, `VendorToPassengerDTO`:
+
+```
+@DefaultSchema(JavaFieldSchema.class)
+class VendorToPassengerDTO {
+
+    @JsonProperty(value = "PassengerCount")
+    Integer PassengerCount;
+    
+    @JsonProperty(value = "VendorID")
+    Integer VendorID;
+
+    @SchemaCreate
+    public VendorToPassengerDTO(Integer passengerCount, Integer vendorID) {
+        this.PassengerCount = passengerCount;
+        this.VendorID = vendorID;
+    }
+
+    // Function for TypeDescription
+    public static VendorToPassengerDTO of(final Integer passengerCount, final Integer vendorID) {
+        return new VendorToPassengerDTO(passengerCount, vendorID);
+    }
+
+    // Setter
+    // Getter
+    // ToString
+}
+```
+
+`Pipeline` can't use select, group, and so on, because it doesn't understand the data structure, so we need to write our own `Coder`:
+
+```
+class CustomCoderSecond extends Coder<VendorToPassengerDTO> {
+    final ObjectMapper objectMapper = new ObjectMapper();
+
+    private static final CustomCoderSecond INSTANCE = new CustomCoderSecond();
+
+    public static CustomCoderSecond of() {
+        return INSTANCE;
+    }
+
+    @Override
+    public void encode(VendorToPassengerDTO dto, OutputStream outStream) throws IOException {
+        final String result = dto.toString();
+        outStream.write(result.getBytes());
+    }
+
+    @Override
+    public VendorToPassengerDTO decode(InputStream inStream) throws IOException {
+        final String serializedDTOs = new String(StreamUtils.getBytesWithoutClosing(inStream));
+        return objectMapper.readValue(serializedDTOs, VendorToPassengerDTO.class);
+    }
+
+    @Override
+    public List<? extends Coder<?>> getCoderArguments() {
+        return Collections.emptyList();
+    }
+
+    @Override
+    public void verifyDeterministic() {
+    }
+}
+```
+
+### Playground exercise
+
+In the playground window you can find examples of using `Coder`. By running this example, you will see user statistics in certain games. Can you change its structure so that an **additional field** appears if the point is greater than **10** to supply `true`.
```

Comment:
Could you pleas re-phrase the user challenge? I'm not sure what we ask user to do